### ISSUE TITLE
Show travellers' position in the map.

### ISSUE THREAD

**Issue by mrkvon** [Trustroots/trustroots#358]
This is a feature idea. Possibly as an independent service using the future [TR oAuth](https://github.com/Trustroots/trustroots/issues/186).
**Problem:** Travelers can find people who have homes, but don't know about each other (there is no easy way to find people who can't host (only going through contacts)).
**Solution:** Traveling people could appear as a (blue) dot on the map. Kind of like [thereiam.org](http://thereiam.org/), but inside trustroots. It could be filtered out if not wanted.
This would enable travelers to meet. It would even make inviting possible.

_Issue:_ People don't update their position and map becomes useless. (thereiam imho suffers from this)
_Solution:_ Dot stays visible only for a ?week and then disappears. User is notified about it and offered to update it.

I guess this is totally not a necessity now. I'll be probably interested in coding this, but i need to finish other work first.

**Comment by simison** [comment]
Totally. We've got this on our minds since the beginning (#28, #84). Actually already at early times of Hitchwiki maps. 

It's by no means simple feature and there are a lot of issues to solve, but a lot of it gets easier once we have native mobile app (#301).

I'll close that another issue about this same feature since since this is more elaborate.

I'd assume this will be getting closer to our todo-lists after summer.

**Comment by mrkvon** [comment]
Great! And sorry for duplicating.

**Comment by simison** [comment]
No worries! :-) Gimme a shout out if you'd like to work on this at some point and we can see how it should be implemented.

Also related #359

**Comment by keith24** [comment]
https://github.com/Tracman lg-Hi friends.  I made [an app](https://github.com/Tracman-org/Server) for tracking with geolocation and socket.io.  Some of the code could probably be used.  

The dashboard (views/dashboard.html) has a button to set (or track) your location.  The map (views/trac.html) updates automatically when it receives websockets from server.js.  There's also a buggy [Android app](https://github.com/Tracman-Org/Android) which can update location in the background.

**Comment by simison** [comment]
@keith24 neat! Could I get an account to give it a test? :-) Would love to see how you solved UX around this.

**Comment by mrkvon** [comment]
So you actually consider automatic tracking the travelers with geolocation? My thoughts were limited to manual updating so far.
What do you imagine/would you like to see?
What about tracking vs. privacy?
What do you think?

**Comment by keith24** [comment]
OK, yeah, maybe tracking isn't really necessary.  Tracking _could_ be done with `socket.io` and `geolocation.watchPosition()`, as I did.  But just updating location would be easier with AJAX and [`geolocation.getCurrentPosition()`](https://developers.google.com/maps/documentation/javascript/examples/map-geolocation).  

Doesn't seem worth setting up websockets if we don't need them.  But tracking would be neat, because the user's movement down a highway would be visible.  I don't think it's less secure to track users more accurately (or faster, as with websockets), so long as they explicitly opt-in **when** and **while** they want to be tracked.   

I've only tested on chrome for android, but my phone stops tracking when the broswer tab is moved to the background or closed or the phone goes to sleep.   It should just be clear to the user that their GPS location is being published while the tab is in the foreground.  

A more important privacy question is probably: should a user's "last-seen" location be stored in the database (as in Tracman), and for how long?

**Comment by mrkvon** [comment]
Yes, these are interesting points. The automatic getting current position is an interesting idea. It could save a lot of annoying map navigating.

The tracking will probably be a topic for further discussion.

Some issues i can think of:
- I can imagine being evil and downloading some chosen peoples' location regularly through the api.
- How would randomization of the user's location work? It works well for houses updated rarely. However by averaging a few randomly randomized locations updated in a short period of time, one can get the traveler's location very (very) precisely. You keep updating your location while setting your tent in the evening and with simple statistics i can visit you at night.

It could be enough to set one's location with an uncertainty of a few kilometers and not too often. Anything more precise may be a privacy and security issue. Should it be offered?

Let's talk about it (now or later). What is the purpose of this feature? What were the ideas already discussed? Possible issues? What do we want and what tools should we use?

**Comment by simison** [comment]
@mrkvon's brainstorming notes:


>**1. Motivation**
>
>So far the only connection doable via trustroots is a traveler writing to a person offering a hosting place.
>With this feature a traveler would be able to find another traveler nearby, a host can find travelers around, see their needs and host/meet/help them.
>
>**2. Basic implementation**
>
>Show travelers' position on a map as a [blue] dot.
>
>**2.1 Map**
>
>Show a blue dot on a map where a traveler set her position.
>The uncertainty of the position may need to be more than Accommodation for safety (and exact location doesn't matter that much anyways). (1-2 km?)
>When clicking the blue dot on map, show info about traveler in similar fashion to accomodation:
>one can write what she does there, whether he is looking for a host, staying till when, wanna meet, need laundry, cooking etc.
>
>there should be a checkbox to show/hide travelers in the map.
>
>**2.1.1 Prevent the abandoned dots**
>dots would expire after a specific time.
>i.e.
>the first ?three days after setting/refreshing the position the dot is dark blue.
>next four days it goes lighter and lighter
>after a ?week it disappears if not refreshed.
>
>**2.2 Setting a position**
>
>Map similar to setting Accommodation for setting a Location.
>Free field for writing about the traveler's whatever.
>When 
>
>**2.3 Email notification**
>
>some time before disappearing of the dot, the email notification might be sent offering to refresh or relocate the position.
>
>The message could go like this:
>
>_Are you still in ... ?_
>_You can refresh your location if you're still around._
>_Refresh, Remove, Manage_
>or
>_I'll stay longer, I already left, Manage_
>_You can ignore this message. In that case your current location will disappear (when)._
>
>_Refresh_ - the link will refresh the current position (the api link + username + unique hash code)
>_Remove_ - the link will remove the position
>_Manage_ - link to the managing of Traveller Location (the Setting position page)
>
>**3. Future and uncertainties**
>
> * to keep or not to keep the history of locations in database?
> * the location might be set manually or with geolocation
> * it might be better not to update the location automatically by geolocation. it could compromise the uncertainty of traveler's location. Averaging a few nearby fuzzy locations updated while camping would precisely show where the traveler is.
> * n future there might be a notification to people who want to receive info about travelers around them. not needed now.
> * should it be possible to set Accommodation and Location at the same time?
>

**Comment by simison** [comment]
There are quite a few other things related to this so before any further brainstorming, I'll have to put this in place regarding everything else. Mostly #359, #396 and #395

**Comment by simison** [comment]
First part of this is done via _meet offers_: https://github.com/Trustroots/trustroots/pull/543 :tada:

**Comment by hitchhooker** [comment]
Being victim of involuntary tracking by entities I dont like, idea of voluntary location tracking to assist me connect with people gets me really fired up. 
I would be up for using real time tracking @keith24 brought up earlier out of usability and comfort over manual updating, but location sharing should definitely be voluntary option with filters to who it's shown(no-one/friends/tribes/everyone?), when/while and time interval+mixer of location update.

**Comment by noahsmindfuck** [comment]
We get (mostly CS-dropout's) requests for a feature to see who is travelling for hosts so that they could chose.
Maybe that can be solved with this feature, too.
